File: Data Structures & Algorithms/reverse-bits/submission-0.py
class Solution:
    def reverseBits(self, n: int) -> int:
        count = 0
        ans = 0
        while n > 0:
            ans = ans * 2 + (n % 2)
            n //= 2
            count += 1
        ans = ans << (32 - count)
        return ans
        

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        int n = nums.size();
        unordered_map<int, int> count;
        vector<vector<int>> freq(n+1);
        for (int val : nums) {
            count[val] = 1 + count[val];
        }
        for (const auto& it : count) {
            freq[it.second].push_back(it.first);
        }

        vector<int> res;
        for (int i = n; i > 0; i--) {
            for (int v : freq[i]) {
                res.push_back(v);
                if (res.size() == k) {
                    return res;
                }
            }
        }
        return res;
    }
};
